### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by kichristensen**
Add support for building Unicorn engine on Windows

### CODE REVIEW

File: libunicorn-sys/build.rs [hunk 1]
```diff
@@ -5,11 +5,18 @@ extern crate pkg_config;
 use std::path::Path;
 use std::process::Command;
 use std::env;
+use std::io::ErrorKind;
```
Comment (ekse) [inline libunicorn-sys/build.rs]: Unused imports?

File: libunicorn-sys/build.rs [hunk 2]
```diff
@@ -5,11 +5,18 @@ extern crate pkg_config;
...
+use std::fs;
 
 fn main() {
-    if Command::new("pkg-config").output().is_ok()
-        && pkg_config::Config::new().atleast_version("1.0.0").probe("unicorn").is_ok() {
-        return;
+    let target = env::var("TARGET").unwrap();
+    let windows = target.contains("windows");
```
Comment (ekse) [inline libunicorn-sys/build.rs]: This line can be removed.

File: libunicorn-sys/build.rs [hunk 3]
```diff
@@ -25,10 +32,39 @@ fn main() {
...
+        let unicorn = "libunicorn.a";
+        let _ = Command::new("cp").current_dir("unicorn").arg(&unicorn).arg(&out_dir).status();
+        
+        println!("cargo:rustc-link-search=native={}", out_dir);
+        println!("cargo:rustc-link-lib=static=unicorn");
+    } else {
+        let libpath = env::var("LIB").unwrap();
+        env::set_var("LIB", libpath + ";" + &out_dir);
+        
+        let platformToolSet = match env::var("PLATFORMTOOLSET") {
```
Comment (ekse) [inline libunicorn-sys/build.rs]: Please rename variable to `platform_tool_set` to avoid warning.

**Comment by ekse** [comment]
I prepared a branch with the fixes and I also used the build-helper crate to simplify the linking code. Could you test that it works fine on Windows?

https://github.com/ekse/unicorn-rs/commit/c94cb48926a2a7443408732d3306a4795af46e2e

**Comment by kichristensen** [comment]
Sure, I'll test it

**Comment by kichristensen** [comment]
@ekse Of some reason it doesn't work anymore, neither my version or yours. It work when building through Visual Studio but not from the command line. I'll try to figure it out

**Comment by kichristensen** [comment]
Seems like it is a problem with the samples, which was built by mistake, it works if you change line 54 from `.args(&["msvc/unicorn.sln", &properties])` to `.args(&["msvc/unicorn.sln", "/t:unicorn_static", &properties])`.

Also line 63 has to be change from ´link_lib(Some(build_helper::LibKind::Static), "unicorn");´ to `link_lib(Some(build_helper::LibKind::Static), "unicorn_static");`

**Comment by ekse** [comment]
Thanks, I applied your changes, could you check one last time and then I'll merge.

**Comment by kichristensen** [comment]
Yes, I'll check it tonight

**Comment by kichristensen** [comment]
It works like a charm

**Comment by ekse** [comment]
The code has been merged and I pushed version 0.7.0 on crates.io. Thank you for your help!

**Pull request merged**